File: src/main/java/com/wikia/tibia/repositories/CreatureRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.CreatureGateway;
import com.wikia.tibia.objects.Creature;

public class CreatureRepository extends WikiObjectRepository {

    public CreatureRepository() {
        super(Creature.class, new CreatureGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/CreatureRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Creature

class CreatureRepository : WikiObjectRepository<Creature>(
    wikiObjectClass = Creature::class.java,
    wikiObjectGateway = WikiObjectGateway(Contract.CREATURES)
)
